api/routes/auth.js

```diff
@@ -0,0 +1,153 @@
+const router = require('express').Router()
+const bcrypt = require('bcrypt');
+const { sign, verify } = require('jsonwebtoken');
+const User = require('../models/user')
+const { SECRET_KEY } = process.env
+
+const generateToken = (id) => {
+    const payload = { id }
+    const options = { expiresIn: '1 day' }
+    return sign(payload, SECRET_KEY, options)
+}
+
+router.use(async (req, res, next) => {
+    try{
+      delete req.user
+      const token = req.headers.authorization.split('Bearer ')[1]
+      const payload = verify(token, SECRET_KEY)
+      const user = await User.findOne({ _id: payload.id }).select('-__v -password')
+      if(user){
+        req.user = user
+      }
+    }finally{
+      next()
+    }
+  })
```

Comment:
This is great! Just be careful where to put this. It's likely that not _all_ routes will need auth. For example, your signup and login routes below will still attempt to run this code, even though it's not needed. 

Additionally, this could be put in `app.js`.